packages/vite/src/node/shortcuts.ts

```diff
@@ -0,0 +1,136 @@
+import colors from 'picocolors'
+import type { ViteDevServer } from './server'
+import { openBrowser } from './server/openBrowser'
+import type { HmrOptions } from './server/hmr'
+import { isDefined } from './utils'
+
+export type BindShortcutsOptions = {
+  /**
+   * Print a one line hint to the terminal.
+   */
+  print?: boolean
+  additionalShortCuts?: (CLIShortcut | undefined | null)[]
+}
+
+export type CLIShortcut = {
+  key: string
+  description: string
+  action(server: ViteDevServer): void | Promise<void>
+}
+
+export function bindShortcuts(
+  server: ViteDevServer,
+  opts: BindShortcutsOptions,
+): void {
+  if (!server.httpServer) return
+
+  if (opts.print) {
+    server.config.logger.info(
+      colors.dim(colors.green('  ➜')) +
+        colors.dim('  press ') +
+        colors.bold('h') +
+        colors.dim(' to show help'),
+    )
+  }
+
+  const shortcuts = (opts.additionalShortCuts ?? [])
+    .filter(isDefined)
+    .concat(BASE_SHORTCUTS)
```

Comment:
Should these be sorted alphabetically?

Reply:
I think there is a quit to having exit at the end, for the rest I have no opinions.

Unresolved question is what to do when there is an override. Here it's in the middle because both are displayed but the custom one take precedence

Reply:
Throw an error means that adding a shortcut would become a breaking change :/

Reply:
Ah good point. Definitely **don't** throw an error 😆